src/server/pachyderm_test.go

```diff
@@ -2138,8 +2138,11 @@ func TestUpdatePipelineRunningJob(t *testing.T) {
 		if len(jobInfos) != 1 {
 			return fmt.Errorf("wrong number of jobs")
 		}
-		if pps.JobState_JOB_RUNNING != jobInfos[0].State {
-			return fmt.Errorf("wrong state: %v for %s", jobInfos[0].State, jobInfos[0].Job.ID)
+
+		state := jobInfos[0].State
+
+		if state != pps.JobState_JOB_RUNNING && state != pps.JobState_JOB_MERGING {
+			return fmt.Errorf("wrong state: %v for %s", state, jobInfos[0].Job.ID)
```

Comment:
This doesn't guarantee that the job enters state merging, but I think that'd be pretty annoying to do in a test and would be flaky, so I'm cool with this.